pkg/kubelet/stats/pidlimit/pidlimit_linux.go

```diff
@@ -39,12 +41,36 @@ func Stats() (*statsapi.RlimitStats, error) {
 		}
 	}
 
-	var info syscall.Sysinfo_t
-	syscall.Sysinfo(&info)
-	procs := int64(info.Procs)
-	rlimit.NumOfRunningProcesses = &procs
+	// Prefer to read "/proc/loadavg" when possible because sysinfo(2)
+	// returns truncated number when greater than 65538. See
+	// https://github.com/kubernetes/kubernetes/issues/107107
+	if procs, err := runningTaskCount(); err == nil {
+		rlimit.NumOfRunningProcesses = &procs
+	} else {
+		var info syscall.Sysinfo_t
+		syscall.Sysinfo(&info)
+		procs := int64(info.Procs)
+		rlimit.NumOfRunningProcesses = &procs
+	}
 
 	rlimit.Time = v1.NewTime(time.Now())
 
 	return rlimit, nil
 }
+
+func runningTaskCount() (int64, error) {
+	// Example: 1.36 3.49 4.53 2/3518 3715089
+	bytes, err := ioutil.ReadFile("/proc/loadavg")
```

Comment:
let's have an e2e test for this functionality so we can make sure to catch potential issues with the format change.

Reply:
There is an e2e test for pid eviction https://github.com/kubernetes/kubernetes/blob/8a259641532d12f730d0fc6b237d36206d405e52/test/e2e_node/eviction_test.go#L448. I think it should be able to catch the format change.